BUG False positive removal on windows due to IOError

First of all thanks for yet another great tool 😄 
Today I was adding `pre-commit` (one hook at a time) to an existing codebase and when I added `yesqa`, it started having a nasty fight with the `flaek8` hook. When I ran `pre-commit run --all` on linux (WSL) there was no problem, so I did some digging.

When I started printing the responses from `flake8`, I found that all false positive removals returned `0       E902`, which [corresponds to an IOError](https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes).

I added:

```python
        print(filename)
        print(line)
```
to:
https://github.com/asottile/yesqa/blob/7a009f3ee493c796827ee334f9058b110a0e0db8/yesqa.py#L25-L30

My guess is that this is due to [`NamedTemporaryFile` only being partially supported by windows](https://docs.python.org/3.8/library/tempfile.html#tempfile.NamedTemporaryFile).
https://github.com/asottile/yesqa/blob/7a009f3ee493c796827ee334f9058b110a0e0db8/yesqa.py#L121-L128

My local quick and dirty hotfix for now is adding another check, like the one you use:
https://github.com/asottile/yesqa/blob/7a009f3ee493c796827ee334f9058b110a0e0db8/yesqa.py#L130-L132

```python
    if any('E902' in v for v in flake8_results.values()):
        print(f'{filename}: IO error (skipping)')
        return 0
```

P.S.: Not sure where `E999` comes from, but  judging by the error message this might be a typo/the error code changed and you might have meant to write [`E901`](https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes)

it works fine???

```
(venv) C:\Users\Anthony\workspace\babi>python
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tempfile
>>> import os.path
>>> import subprocess
>>> with tempfile.TemporaryDirectory() as tmpdir:
...     fname = os.path.join(tmpdir, 'f.py')
...     with open(fname, 'w') as f:
...         f.write('import os')
...     import sys
...     subprocess.call((sys.executable, '-mflake8', fname))
...
9
C:\Users\Anthony\AppData\Local\Temp\tmpkn1986mo\f.py:1:1: F401 'os' imported but unused
C:\Users\Anthony\AppData\Local\Temp\tmpkn1986mo\f.py:1:10: W292 no newline at end of file
1
```

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -16,5 +16,9 @@ jobs:
 - template: job--pre-commit.yml@asottile
 - template: job--python-tox.yml@asottile
   parameters:
+    toxenvs: [py37]
+    os: windows
+- template: job--python-tox.yml@asottile
+  parameters:
     toxenvs: [pypy3, py36, py37, py38]
     os: linux

--- a/yesqa.py
+++ b/yesqa.py
@@ -118,14 +118,17 @@ def fix_file(filename: str) -> int:
     if src_no_comments == contents_text:
         return 0
 
-    with tempfile.NamedTemporaryFile(
+    fd, path = tempfile.mkstemp(
         dir=os.path.dirname(filename),
         prefix=os.path.basename(filename),
         suffix='.py',
-    ) as tmpfile:
-        tmpfile.write(src_no_comments.encode())
-        tmpfile.flush()
-        flake8_results = _run_flake8(tmpfile.name)
+    )
+    try:
+        with open(fd, 'wb') as f:
+            f.write(src_no_comments.encode())
+        flake8_results = _run_flake8(path)
+    finally:
+        os.remove(path)
 
     if any('E999' in v for v in flake8_results.values()):
         print(f'{filename}: syntax error (skipping)')

